arch/risc-v/src/k210/k210_head.S

```diff
@@ -223,9 +223,16 @@ normal_irq:
   .type   g_intstackalloc, object
   .type   g_intstackbase, object
 g_intstackalloc:
-  .skip  (((CONFIG_ARCH_INTERRUPTSTACK * 2) & ~7))
+#ifndef CONFIG_SMP
+  .skip  ((CONFIG_ARCH_INTERRUPTSTACK + 4) & ~7)
+#else
+  .skip  (((CONFIG_ARCH_INTERRUPTSTACK * CONFIG_SMP_NCPUS) + 4) & ~7)
+#endif
 g_intstackbase:
-  .skip  8
-  .size  g_intstackbase, 8
-  .size  g_intstackalloc, ((CONFIG_ARCH_INTERRUPTSTACK * 2) & ~7)
+  .size  g_intstackbase, 0
```

Comment:
@masayuki2009 I think we only need to backport the `.size` and `.skip` of `g_intstackalloc`, right? The other changes are fixes to side-effects from PR https://github.com/apache/incubator-nuttx/pull/3471 which will not be backported.

Lines 228 and 229 could be left as is.

Reply:
@gustavonihei I understand that PR #3471 will not be backported to release/10.1. However, I noticed that PR #3632 (i.e. bp-3614) has the same side effects introduced in  PR #3471. So I think this PR should be merged before applying PR#3632.

Reply:
Agreed, so let's merge this one first.